packages/twenty-front/src/modules/ui/input/components/IconPicker.tsx

```diff
@@ -116,6 +116,10 @@ export const IconPicker = ({
         })
       : [];
 
+    const filteredSearchResults = filteredIconKeys.filter(
```

Comment:
const isSelectedIconMatchingFilter = filterIconKeys.find(...)

then we can change:
const uniqueFilteredIconKeys = [
      ...new Set(
        selectedIconKey
          ? [selectedIconKey, ...filteredIconKeys]
          : filteredIconKeys,
      ),
    ];
to only prepend the selectedIconKey if it maches